packages/contracts/src/strategies/CTokenBaseStrategy.sol

```diff
@@ -151,9 +152,14 @@ abstract contract CTokenBaseStrategy is ICInterestRate, BaseStrategy {
         }
 
         (uint256 sharesAfter, uint256 underlyingAfter) = depositedBalanceSnapshot();
-        emit WithdrawnFromProtocol(amountToRedeem, sharesBefore, deposits, sharesAfter, underlyingAfter);
 
-        return amountToRedeem;
+        // underlying protocol can return less than calculated above, so this check is required to avoid manipulations
+        uint256 balanceAfter = _underlyingBalance();
+        uint256 redeemedAmount = balanceAfter - balanceBefore;
+
+        emit WithdrawnFromProtocol(amountToRedeem, sharesBefore, deposits, sharesAfter, underlyingAfter, redeemedAmount);
+
+        return redeemedAmount;
```

Comment:
Please add test for this logic change

Reply:
Do we really need this change?
`balanceAfter - balanceBefore` is not a redeemed amount in fact. What if there was some amount of assets on the contrac already? Or, someone send some to the contract address, so then it will produce wrong value. 

Redeemed amount is the actual amount of assets that was withdrawed from the underlying contract (ApeSwap), i.e. `redeemedAmount` = `amountToRedeem` as it was before (otherwise transaction will be reverted)

Reply:
This is correct logic, just what we requested to redeem not equal what was redeemed at the end. We need not trust third party contracts for this and check by self

Reply:
added additional fraud test